source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -505,15 +505,14 @@ ValidationResults ContextImpl::customVerifyCertChain(
     extended_socket_info->setCertificateValidationStatus(Ssl::ClientValidationStatus::NotValidated);
```

Comment:
Either `extended_socket_info->setCertificateValidationStatus()` needs to be called after `doVerifyCertChain()` returns non-Pending status as well, or both places should defer calling the setter to the caller of this method.

Reply:
The fact that this is not breaking any test makes me wonder if anyone cares about the client validation status at all.

Reply:
> The fact that this is not breaking any test makes me wonder if anyone cares about the client validation status at all.

Yeah, I wondered about this as well. The PlatformBridge validator also wasn't calling `setCertificateValidationStatus()` and everything passed. I wonder if people only use the status in the failure cases, and we don't have broad test coverage of the failure cases?

> Either extended_socket_info->setCertificateValidationStatus() needs to be called after doVerifyCertChain() returns non-Pending status as well

Good point. Done.